File: server/app/schemas/ip_sticker.py
"""
IP 贴纸相关 Pydantic 模式
"""

from datetime import datetime

from pydantic import BaseModel, Field


class SessionItem(BaseModel):
    """会话列表项"""
    session_id: str
    status: str
    current_step: int
    source_image_id: str | None = None
    created_at: datetime
    updated_at: datetime


class SessionDetail(BaseModel):
    """会话详情（含消息历史）"""
    session_id: str
    user_id: str
    status: str
    current_step: int
    source_image_id: str | None = None
    created_at: datetime
    updated_at: datetime
    messages: list["MessageItem"] = Field(default_factory=list)
    master_template: "MasterTemplateItem | None" = None
    stickers: list["StickerItem"] = Field(default_factory=list)


class MessageItem(BaseModel):
    """聊天消息"""
    message_id: str
    role: str
    message_type: str
    content: str | None = None
    images: list[dict] | None = None
    actions: list[dict] | None = None
    sequence: int
    created_at: datetime


class MasterTemplateItem(BaseModel):
    """IP 母版"""
    template_id: str
    master_image_url: str
    master_thumbnail_url: str | None = None
    character_description: str | None = None
    version: int
    is_locked: bool
    created_at: datetime


class StickerItem(BaseModel):
    """贴纸结果"""
    sticker_id: str
    sticker_index: int
    label: str
    result_url: str
    thumbnail_url: str | None = None
    status: str
    batch_type: str
    is_favorite: bool
    redraw_count: int
    created_at: datetime


class CreateSessionRequest(BaseModel):
    """创建会话请求"""
    image_id: str | None = Field(None, description="源图片ID（可选，也可后续通过 WS 设置）")


class SendMessageRequest(BaseModel):
    """发送消息请求（REST 降级方案）"""
    type: str = Field(description="消息类型")
    payload: dict = Field(default_factory=dict, description="消息内容")


class SessionListResponse(BaseModel):
    """会话列表响应"""
    sessions: list[SessionItem]

File: web/src/types/ipSticker.ts
// IP 贴纸聊天相关类型定义

/** 客户端消息类型 */
export type ClientMessageType =
  | 'set_photo'
  | 'chat'
  | 'confirm_base'
  | 'modify_base'
  | 'approve_test'
  | 'regenerate_test'
  | 'generate_full'
  | 'redraw_sticker'
  | 'toggle_favorite'

/** 服务端消息类型 */
export type ServerMessageType =
  | 'session_created'
  | 'session_resumed'
  | 'chat_reply'
  | 'image_started'
  | 'image_completed'
  | 'image_grid_completed'
  | 'state_changed'
  | 'action_required'
  | 'toggle_favorite_done'
  | 'sticker_updated'
  | 'export_ready'
  | 'error'

/** 客户端 → 服务端消息 */
export interface ClientMessage {
  type: ClientMessageType
  payload: Record<string, unknown>
  request_id?: string
}

/** 服务端 → 客户端消息 */
export interface ServerMessage {
  type: ServerMessageType
  payload: Record<string, unknown>
  request_id?: string
  timestamp: string
}

/** 图片项 */
export interface ChatImage {
  url: string
  thumbnail_url?: string
  sticker_id?: string
  template_id?: string
  label?: string
  index?: number
  status?: string
  error?: string
}

/** 操作按钮 */
export interface ChatAction {
  action: string
  label: string
  type?: 'primary' | 'default' | 'danger'
  payload?: Record<string, unknown>
}

/** 显示用的聊天消息（从 ServerMessage 转换） */
export interface DisplayMessage {
  id: string
  role: 'user' | 'assistant' | 'system'
  type: string
  content?: string
  images?: ChatImage[]
  actions?: ChatAction[]
  timestamp: string
}

/** IP 母版信息 */
export interface MasterTemplate {
  templateId: string
  masterImageUrl: string
  masterThumbnailUrl?: string
  characterDescription?: string
  version: number
  isLocked: boolean
  createdAt: string
}

/** 贴纸结果 */
export interface StickerItem {
  stickerId: string
  stickerIndex: number
  label: string
  resultUrl: string
  thumbnailUrl?: string
  status: string
  batchType: string
  isFavorite: boolean
  redrawCount: number
  createdAt: string
}

/** 会话列表项 */
export interface SessionItem {
  session_id: string
  status: string
  current_step: number
  source_image_id?: string
  created_at: string
  updated_at: string
}

/** 会话详情 */
export interface SessionDetail {
  sessionId: string
  userId: string
  status: string
  currentStep: number
  sourceImageId?: string
  createdAt: string
  updatedAt: string
  messages: MessageItem[]
  masterTemplate?: MasterTemplate
  stickers: StickerItem[]
}

/** 消息项（REST API 返回） */
export interface MessageItem {
  messageId: string
  role: string
  messageType: string
  content?: string
  images?: ChatImage[]
  actions?: ChatAction[]
  sequence: number
  createdAt: string
}
